pulsar-broker/src/main/java/org/apache/pulsar/broker/service/ServerCnx.java

```diff
@@ -440,13 +440,19 @@ protected void handleLookup(CommandLookupTopic lookup) {
 
         final Semaphore lookupSemaphore = service.getLookupRequestSemaphore();
         if (lookupSemaphore.tryAcquire()) {
-            if (invalidOriginalPrincipal(originalPrincipal)) {
-                final String msg = "Valid Proxy Client role should be provided for lookup ";
-                log.warn("[{}] {} with role {} and proxyClientAuthRole {} on topic {}", remoteAddress, msg, authRole,
-                        originalPrincipal, topicName);
-                ctx.writeAndFlush(newLookupErrorResponse(ServerError.AuthorizationError, msg, requestId));
+            try {
+                if (invalidOriginalPrincipal(originalPrincipal)) {
+                    final String msg = "Valid Proxy Client role should be provided for lookup ";
```

Comment:
Hi, I don't think `invalidOriginalPrincipal` will throw exception.

Reply:
ahaha, I think that other people may ignore the "finally release  semaphore" when they change the code in the future, so submit this PR